jep/0000/README.adoc

```diff
@@ -0,0 +1,227 @@
+
+One of the four pillars of Jenkins Essentials is that it should be link:https://github.com/jenkinsci/jep/tree/master/jep/300#sane-defaults[autoconfigured with sane defaults for various cloud platforms]. This specification explains what setup is done to autoconfigure Jenkins Essentials to run in the link:https://aws.amazon.com[Amazon Web Service] environment.
+
+== Specification
+
+=== Global overview
+
+The setup uses an link:https://aws.amazon.com/cloudformation[AWS _CloudFormation_ template] that instantiates the right resources.
+
+It creates an EC2 instance, to host the Jenkins Essentials master in a dedicated Instance Profile.
+The master will be able to provision agents from EC2, and store artifacts directly in S3.
+
+=== Used plugins and configuration
+
+* link:https://plugins.jenkins.io/ec2[EC2 plugin]: to provide dynamically provisioned agents
+  footnote:[even more important for Jenkins Essentials which is configured with the link:JENKINS-49861[sane 0-executor for the master node]].
+* link:https://plugins.jenkins.io/artifact-manager-s3[Artifact Manager on S3 plugin]: to store artifacts and Pipeline stashes in Amazon S3.
+
+// JENKINS-52342 for ^ which is the design, but not done fully yet.
+
+As a general rule for Essentials, we are using the `configuration-as-code` plugin to do the necessary configuration.
+
+*No credentials* are specified.
+The permissions are retrieved dynamically through the instance profile used when creating the EC2 instance for the Essentials master.
+
+=== Instance Profile and IAM role
+
+A dedicated instance profile is created by the _CloudFormation_ template to allow creating EC2 instances for agents, and access a specific S3 bucket to store artifacts and Pipeline stashes.
+
+==== EC2 policy
+
+[source,json]
+"Statement": [{
+    "Sid": "ControlEC2JenkinsAgents",
+    "Effect": "Allow",
+    "Action": [
+        "ec2:DescribeInstances",
+        "ec2:TerminateInstances",
+        "ec2:RequestSpotInstances",
+        "ec2:DeleteTags",
+        "ec2:CreateTags",
+        "ec2:DescribeRegions",
+        "ec2:RunInstances",
+        "ec2:DescribeSpotInstanceRequests",
+        "ec2:StopInstances",
+        "ec2:DescribeSecurityGroups",
+        "ec2:GetConsoleOutput",
+        "ec2:DescribeImages",
+        "ec2:CancelSpotInstanceRequests",
+        "ec2:StartInstances",
+        "ec2:DescribeAvailabilityZones",
+        "ec2:DescribeSubnets",
+        "ec2:DescribeKeyPairs"
+    ],
+    "Resource": "*"
+}]
+
+==== S3 bucket policy
+
+FIXME: refine through JENKINS-52342.
+
+[source,json]
+"Statement": [{
+        "Sid": "TodoRefineSecurityALot",
+        "Effect": "Allow",
+        "Action": [
+            "s3:PutObject",
+            "s3:GetObject",
+            "s3:ListBucket",
+            "s3:DeleteObject"
+        ],
+        "Resource": "*"
+    }
+]
+
+=== Security groups
+
+==== Master
+A dedicated group is created to allow access to the following ports:
+
+22:: restricted to an optional parameter.
```

Comment:
So by default, no CLI over SSH, nor Git server for (old-style) `workflowLibs.git`, `userContent.git`, etc.

Reply:
> nor Git server for (old-style) workflowLibs.git

I _think_ the "old-style" part of the message sums it up. As per the [_Obvious Path To User Success_](https://github.com/jenkinsci/jep/tree/master/jep/300#obvious-path) pillar, I think we shouldn't enable this.

Or do you think there are still use cases that would be important for "modern" usages nowadays, and easy enough for the [target audience](https://github.com/jenkinsci/jep/tree/master/jep/300#target-audience) of Essentials?